lib/cereal/grpc/grpc.go

```diff
@@ -80,9 +82,30 @@ func (c *workflowCompleter) EnqueueTask(task *backend.Task, opts backend.TaskEnq
 	c.tasks = append(c.tasks, t)
 	return nil
 }
+func (c *workflowCompleter) finish(err error) error {
+	defer c.s.CloseSend() // nolint: errcheck
```

Comment:
Curious why we need to close the sender end of the stream twice?

Reply:
all the paths don't explicitly call close send. defering makes sure we always do

Reply:
I suppose if that's the worry we could close it before the first err return and wouldn't require the defer to an already closed stream on the happy path.